django_cloud_deploy/tests/lib/test_base.py

```diff
@@ -100,16 +100,112 @@ def setUp(self):
+            # more accounts in the remaining pages.
+            accounts += [
+                account['email'] for account in response.get('accounts', [])
+            ]
+            if 'nextPageToken' in response:
+                request = service.projects().serviceAccounts().list(
+                    name=resource_name, pageToken=response.get('nextPageToken'))
+                response = request.execute()
+            else:
+                break
+        return accounts
+
+    def list_clusters(self, service=None):
+        service = service or discovery.build(
+            'container',
+            'v1',
+            credentials=self.credentials,
+            cache_discovery=False)
+        request = service.projects().zones().clusters().list(
+            projectId=self.project_id, zone=self.zone)
+        response = request.execute()
+        return [
+            cluster.get('name', '') for cluster in response.get('clusters', [])
+        ]
+
+    def list_buckets(self, service=None):
+        service = service or discovery.build(
+            'storage',
+            'v1',
+            credentials=self.credentials,
+            cache_discovery=False)
+        request = service.buckets().list(project=self.project_id)
+        response = request.execute()
+        return [bucket.get('name', '') for bucket in response.get('items', [])]
+
+    def list_enabled_services(self, service_usage_service=None):
+        service_usage_service = service_usage_service or discovery.build(
+            'serviceusage',
+            'v1',
+            credentials=self.credentials,
+            cache_discovery=False)
+        parent = '/'.join(['projects', self.project_id])
+        request = service_usage_service.services().list(
+            parent=parent, filter='state:ENABLED')
+        response = request.execute()
+        return [service['config']['name'] for service in response['services']]
+
+    def list_instances(self, service=None):
+        service = service or discovery.build(
+            'sqladmin',
+            'v1beta4',
+            cache_discovery=False,
+            credentials=self.credentials)
+        request = service.instances().list(project=self.project_id)
+        response = request.execute()
+        instances = [item['name'] for item in response['items']]
+        return instances
+
+    def list_databases(self, instance_name, service=None):
+        service = service or discovery.build(
+            'sqladmin',
+            'v1beta4',
+            cache_discovery=False,
+            credentials=self.credentials)
+        request = service.databases().list(
+            project=self.project_id, instance=instance_name)
+        response = request.execute()
+        databases = [item['name'] for item in response['items']]
+        return databases
+
+
+class ResourceCleanUp(BaseTest):
     """Class for test cases which need resource cleaning up."""
 
+    def _delete_cluster(
```

Comment:
These changes are made because we want to reuse code for resource deletion in integration tests and the newly added resource cleanup.